### PR TITLE
build(codepush): codepush dev menu options

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-181] 

### Description

Code push integration part 2 of 3: https://www.notion.so/Adopt-CodePush-Over-the-air-updates-in-Eigen-cb95950057e34f259c801a3359cefe7c

Depends on: https://github.com/artsy/eigen/pull/9197 which should be merged first. ✅ 

Adds a codepush section to the dev menu to allow running builds from 3 deployments (Staging, Canary, Production)

**Staging:** the latest code in main
**Canary:** a place for devs to deploy in progress work before getting into main
**Production:** empty for now but possibly a place to deploy hot fixes in the future

#### Downloading an update

https://github.com/artsy/eigen/assets/49686530/9a4931d7-b8db-4054-94d4-092f09ef4549

### Viewing the active release

https://github.com/artsy/eigen/assets/49686530/0df72522-2835-4fb1-902e-cbce92dc249f

### Follow ups
- Add ci scripts + flow + sentry integration 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add code push dev menu options - Brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-181]: https://artsyproduct.atlassian.net/browse/PHIRE-181?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0a83b5c3e56a246a9edbf9d231050dd6e416204b
</p>

File: src/app/utils/DevMenu/DevMenu.tsx [hunk 1]
```diff
@@ -26,9 +26,11 @@ import { Versions } from "app/store/migration"
 import { eigenSentryReleaseName } from "app/system/errorReporting//sentrySetup"
 import { dismissModal, goBack, navigate } from "app/system/navigation/navigate"
 import { RelayCache } from "app/system/relay/RelayCache"
+import { CodePushOptions } from "app/utils/DevMenu/CodePushOptions"
+import { useUnleashEnvironment } from "app/utils/experiments/hooks"
 import { useBackHandler } from "app/utils/hooks/useBackHandler"
 import { capitalize, compact, sortBy } from "lodash"
-import { useState } from "react"
+import React, { useState } from "react"
```
Comment (gkartalis) [inline src/app/utils/DevMenu/DevMenu.tsx:33]: 👀 do we need this???

**Review by gkartalis — APPROVED** [review]
Looks great to me! 🔥

File: src/app/utils/DevMenu/CodePushOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import {
```
Comment (damassi) [inline src/app/utils/DevMenu/CodePushOptions.tsx:1]: @briansw @gkartalis - can we move this folder to https://github.com/artsy/eigen/tree/main/src/app/system/devTools? 

Trying to keep a clean / organized separation between our system-level things and the rest of the app, hence the `devTools` folder

File: src/app/system/devTools/DevMenu/CodePushOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+const codePushKeyToDeployment: { [key: string]: CodePushDeployment } = {
+  [stagingKey]: "Staging",
+  [productionKey]: "Production",
+  [canaryKey]: "Canary",
+}
+
+export const CodePushOptions = () => {
+  const [selectedDeployment, setSelectedDeployment] = useState<CodePushDeployment>("Staging")
```
Comment (MounirDhahri) [inline src/app/system/devTools/DevMenu/CodePushOptions.tsx:40]: **Q:** is there a specific reason why we would want this to point always to staging? I would have assumed you would have the initial state pointing to staging **or production**

**Review by MounirDhahri — COMMENTED** [review]
👏

**Pull request merged**